firewood/src/merkle.rs

```diff
@@ -240,6 +310,72 @@ impl<T: TrieReader> Merkle<T> {
         MerkleKeyValueStream::from_key(&self.nodestore, key.as_ref())
     }
 
+    /// Generate a cryptographic proof for a range of key-value pairs in the Merkle trie.
+    ///
+    /// This method creates a range proof that can be used to verify the existence (or absence)
+    /// of a contiguous set of keys within the trie. The proof includes boundary proofs and
+    /// the actual key-value pairs within the specified range.
+    ///
+    /// # Parameters
+    ///
+    /// * `start_key` - The optional lower bound of the range (inclusive).
+    ///   - If `Some(key)`, the proof will include all keys >= this key
+    ///   - If `None`, the proof starts from the beginning of the trie
+    ///
+    /// * `end_key` - The optional upper bound of the range (inclusive).
+    ///   - If `Some(key)`, the proof will include all keys <= this key
+    ///   - If `None`, the proof extends to the end of the trie
+    ///
+    /// * `limit` - Optional maximum number of key-value pairs to include in the proof.
```

Comment:
FYI: The go code supported a maximum network packet size. It was very pedantic on making sure it was the largest proof that would fit into a network packet. IMO we don't need to support this, but we might want to call it out.